public/app/features/dashboard/save_modal.ts

```diff
@@ -50,25 +53,35 @@ export class SaveDashboardModalCtrl {
   message: string;
   max: number;
   saveForm: any;
+  saving: boolean;
   dismiss: () => void;
 
   /** @ngInject */
   constructor(private dashboardSrv) {
     this.message = '';
     this.max = 64;
+    this.saving = false;
   }
 
   save() {
     if (!this.saveForm.$valid) {
       return;
     }
 
+    this.setSaving();
+
     var dashboard = this.dashboardSrv.getCurrent();
     var saveModel = dashboard.getSaveModelClone();
     var options = { message: this.message };
 
     return this.dashboardSrv.save(saveModel, options).then(this.dismiss);
   }
+
+  setSaving() {
+    let saveButton = document.querySelector('#saveButton');
```

Comment:
use ng-class in html template , example ng-class="{'btn--processing': ctrl.isSaving}" 

And the class needs to be more generic so it works for all types & themes, not sure if that is possible or just ticky. Also need a way to cancel isSaving state when errors happen. I wonder how we can do that without adding ugly try catch, and how we can make this more more reusable in other places.